lib/wavefile/chunk_readers/smpl_chunk_reader.rb

```diff
@@ -0,0 +1,118 @@
+module WaveFile
+  module ChunkReaders
+    # Internal
+    class SmplChunkReader < BaseChunkReader    # :nodoc:
+      # Public: Returns the ID of the manufacturer
+      # See the list at https://www.midi.org/specifications-old/item/manufacturer-id-numbers
+      attr_reader :manufacturer_id
+
+      # Public: Returns the ID of the product
+      attr_reader :product_id
+
+      # Public: Returns the length of each sample in nanoseconds
+      attr_reader :sample_duration
+
+      # Public: Returns the MIDI note number of the sample (0-127)
+      attr_reader :unity_note
+
+      # Public: Returns the fraction of a semitone up from the specified MIDI unity note field.
+      # A value of 0x80000000 means 1/2 semitone (50 cents) and a value of 0x00000000 means no fine tuning between semitones.
+      # - https://sites.google.com/site/musicgapi/technical-documents/wav-file-format
+      # Integer
+      attr_reader :pitch_fraction
+
+      # Public: Returns the SMPTE format (0, 24, 25, 29 or 30)
+      attr_reader :smpte_format
+
+      # Public: Returns the SMPTE time offset.
+      # This value uses a format of 0xhhmmssff where hh is a signed value that specifies the number of hours (-23 to 23),
+      # mm is an unsigned value that specifies the number of minutes (0 to 59), ss is an unsigned value that specifies
+      # the number of seconds (0 to 59) and ff is an unsigned value that specifies the number of frames (0 to -1).
+      # - https://sites.google.com/site/musicgapi/technical-documents/wav-file-format
+      attr_reader :smpte_offset
+
+      # Public: Returns the number of loops defined in the sample
+      attr_reader :loop_count
+
+      # Public: Returns anumber of bytes used for additional sampler data
+      attr_reader :sampler_data
+
+      # Public: Returns the loop specifications
+      # Array of Loop objects
+      attr_reader :loops
+
+      def initialize(io, chunk_size)
+        @io = io
+        @chunk_size = chunk_size
+      end
+
+      def read
```

Comment:
Since `read` is defined on the object returned by `Reader#sample_info`, it seems like someone could cause unexpected behavior by doing this:
~~~
reader = Reader.new("file.wav")

# This will replace the data inside the `SmplChunkReader`
# instance with the first bytes of the `data` chunk (which
# will be gibberish)
reader.sample_info.read
~~~

I guess this would be undocumented behavior, but it seems like it would be better to prevent the possibility altogether.

One possible way is to have separate data container and data reader objects, instead of combining them into one object. For example, you could create a `SampleChunk` object which only stores the raw fields (similar to `UnvalidatedFormat` or `Format`). Then, `SampleChunkReader` would be changed to instead create an instance of `SampleChunk`, and no longer contain the fields itself.

For example in `SampleChunkReader`, something like:

~~~
def read
  manufacturer_id = @io.sysread(4).unpack(UNSIGNED_INT_32)[0]
  product_id = @io.sysread(4).unpack(UNSIGNED_INT_32)[0]
  # et cetera

  SampleChunk.new(manufacturer_id: manufacturer_id, product_id: product_id, ...et cetera...)
end
~~~